cmd/run_controller/main.go

```diff
@@ -132,16 +153,21 @@ func main() {
 	}
 	controller := runctl.NewController(factory, controllerOpts)
 
-	klog.V(3).Infof("Create Signal Handlers")
+	klog.V(3).InfoS("Create signal handlers")
 	stopCh := signals.SetupShutdownSignalHandler()
 	signals.SetupThreadDumpSignalHandler()
 
-	klog.V(2).Infof("Start Informer")
+	klog.V(2).InfoS("Start Informer")
 	factory.StewardInformerFactory().Start(stopCh)
 	factory.TektonInformerFactory().Start(stopCh)
 
-	klog.V(2).Infof("Run controller (threadiness=%d)", threadiness)
+	klog.V(2).InfoS("Run pipeline run controller", "threadiness", threadiness)
 	if err = controller.Run(threadiness, stopCh); err != nil {
-		klog.Fatalf("Error running controller: %s", err.Error())
+		klog.ErrorS(err, "Failed to run controller")
+		flushLogsAndExit()
 	}
 }
+
+func flushLogsAndExit() {
+	klog.FlushAndExit(klog.ExitFlushTimeout, 1)
```

Comment:
May be you could move the klog.ErrorS inside this function.
May be rename to 'logFatalS'.

Reply:
klog.Fatal exits the application with 255 error code, using logFatalS meaning the consumer should expect exit code 255 on error. However we want to return with exit code 1 for known and managed error cases. 
I would rather combine it ErrorS and klog's exit operation by renaming it 
`logErrorAndExit(err error, msg string, args ...interface{})`

Reply:
Wrapping of error logging is already reverted in b997850089af6d0e6b1c46e298f3f20af2e786b9. Also it is not supported at the moment in logcheck linter.